### DISCUSSION TITLE
Config and setup for Webdriverio running Edge with IE mode (Internet Explorer)

### DISCUSSION THREAD

**Discussion by troyyer** [webdriverio/webdriverio#12045] [Q&A]
Hi there,

has anyone automated on a windows machine, running Edge with IE Mode (I know...I know....)?
I'm trying to follow some blog posts but not able to get it working 
https://blogs.windows.com/msedgedev/2022/02/01/test-ie-mode-in-microsoft-edge-internet-explorer-driver/
https://learn.microsoft.com/en-us/microsoft-edge/webdriver-chromium/ie-mode?tabs=javascript

i'm currently with the following config 

```
   capabilities: [{
        browserName: 'internet explorer',
        'se:ieOptions': {
            setEdgeChromium: true,
           setEdgePath: '.\\node_modules\\.bin\\edgedriver -port=4444', // doesnt seem to work
        },
    }],
    services: [
        'edgedriver'
    ],
```
on first running the Edgedriver gets downloaded. after the download if i just do a plan Edge test with `    capabilities: [{  browserName: 'MicrosoftEdge'  }],`, the driver will start as expected and run.when trialing out the config for ie mode i seem to need to start the driver 
so i start it on port 4444
```
C:\code\new-sample-test>.\node_modules\.bin\edgedriver -port=4444
```
but when trying to run the test the error i get is 
```
 2024-01-18T22:29:36.841Z ERROR webdriver: Request failed with status 500 due to session not created: session not created: No matching capabilities found
[0-0] 2024-01-18T22:29:36.841Z ERROR webdriver: session not created: session not created: No matching capabilities found
```

using Browserstack i was able to run to find some configurations to make it run, but i also wanted to get it running locally as well
```
    capabilities: [
        {   
            'bstack:options': {
              "idleTimeout": 300,
              "os" : "windows",
              "osVersion": 10,
              "browserVersion": 11,
              "buildName": "IEMode Compat",
              networkLogs: true,
              edgeVersion: "118.0"
            },
            browserName: "internet explorer",
            
            'se:ieOptions': {
              'ie.edgechromium': true,
              'ie.ignoreZoomSetting':true,
            }
          }
    ],
```

in my npm package i have the following 

```
  "@wdio/browserstack-service": "^7.34.0",
        "@wdio/cli": "7.2.0",
        "@wdio/local-runner": "^7.34.0",
        "@wdio/mocha-framework": "^7.33.0",
        "@wdio/spec-reporter": "^7.33.0",
        "wdio-edgedriver-service": "^3.0.3",
        "wdio-video-reporter": "^4.0.5"
```

**Comment by christian-bromann** [comment]
@troyyer Internet Explorer can't be automated with Edgedriver, you will need the IEDriver for it https://www.selenium.dev/documentation/ie_driver_server/

**Comment by troyyer** [comment]
thanks @christian-bromann 
so currently i dont have IE on my machine (it's been removed by the company) and so we go to Edge to open up the app in IE mode. so would IEdriver still be able to automate Edge, or do i need both Edgedriver and IEdriver in the project

**Reply by christian-bromann** [reply]
> so we go to Edge to open up the app in IE mode.

I think this can also only be automated by IEDriver, I am not sure though. I thought this browser finally died 👀 guess I am wrong. I guess I can't really help you as I never automated IE or Edge in IE mode.

**Reply by troyyer** [reply]
thanks @christian-bromann, yeeeeeeaaaaaaaa, when the manual tester showed the application I curled up into a ball in the corner......I'll keep digging around and see how I go with WebdriverIO and the IEdriver and document it here.
Anyone else with recent usage of IEdriver would be greatly appreciated.

**Reply by harpreet1993** [reply]
Hey @troyyer  did you find any solution?

**Reply by troyyer** [reply]
@harpreet1993, sorry I tried a few times locally with no success. Had initially got it working with the cloud vendor browserstack and with my time constraints I didn't have time to trial it again locally.
